spec/rmagick/image/chromaticity_spec.rb

```diff
@@ -1,23 +1,31 @@
 RSpec.describe Magick::Image, '#chromaticity' do
-  it 'works' do
-    image = described_class.new(100, 100)
+  it "returns an assigned chromaticity" do
+    image = build_image
 
-    chrom = image.chromaticity
-    expect { image.chromaticity }.not_to raise_error
-    expect(chrom).to be_instance_of(Magick::Chromaticity)
-    expect(chrom.red_primary.x).to eq(0)
-    expect(chrom.red_primary.y).to eq(0)
-    expect(chrom.red_primary.z).to eq(0)
-    expect(chrom.green_primary.x).to eq(0)
-    expect(chrom.green_primary.y).to eq(0)
-    expect(chrom.green_primary.z).to eq(0)
-    expect(chrom.blue_primary.x).to eq(0)
-    expect(chrom.blue_primary.y).to eq(0)
-    expect(chrom.blue_primary.z).to eq(0)
-    expect(chrom.white_point.x).to eq(0)
-    expect(chrom.white_point.y).to eq(0)
-    expect(chrom.white_point.z).to eq(0)
-    expect { image.chromaticity = chrom }.not_to raise_error
-    expect { image.chromaticity = 2 }.to raise_error(TypeError)
+    red_primary = Magick::Primary.new(70, 200, 0)
+    green_primary = Magick::Primary.new(200, 44, 0)
+    blue_primary = Magick::Primary.new(60, 22, 0)
+    white_point = Magick::Primary.new(77, 43, 0)
+    chromaticity = Magick::Chromaticity.new(red_primary, green_primary, blue_primary, white_point)
+    image.chromaticity = chromaticity
+
+    expect(image.chromaticity.red_primary).to eq(red_primary)
+    expect(image.chromaticity.green_primary).to eq(green_primary)
+    expect(image.chromaticity.blue_primary).to eq(blue_primary)
+    expect(image.chromaticity.white_point).to eq(white_point)
+  end
+
+  it "returns default chromaticity values" do
+    image = build_image
+
+    red_primary = Magick::Primary.new(0.6399999856948853, 0.33000001311302185, 0.029999999329447746)
```

Comment:
Why is the precision for these numbers getting out of whack? Is there a Ruby API that preserves the numbers from C?